src/dataflow/src/render/sources.rs

```diff
@@ -160,6 +160,12 @@ where
     // Tokens that we should return from the method.
     let mut needed_tokens: Vec<Rc<dyn std::any::Any>> = Vec::new();
 
+    // This uid must be unique across all different instantiations of a source
+    let uid = SourceInstanceId {
+        source_id: src_id,
+        dataflow_id,
+    };
```

Comment:
We might want to revisit this soon, with @antiguru input about how best to name a dataflow. I think this, for example, isn't particularly stable (it is the timely dataflow id, which isn't stable across restarts, nor disambiguated across compute instances). I think the recommendation is to use the asserted-unique `GlobalId` of the output, until we get something better.

Reply:
Ah, nvm. If this is just lifted from below, no need to solve this now.